lib/cocoapods/installer/analyzer/podfile_dependency_cache.rb

```diff
@@ -0,0 +1,46 @@
+module Pod
+  class Installer
+    class Analyzer
+      # Caches podfile & target definition dependencies, so they do not need to be re-computed
+      # from the internal hash on each access
+      class PodfileDependencyCache
```

Comment:
Looks like we need tests for this (rn i think it might be covered transitively by the callsites, but explicit tests for from_podfile would be nice)